### ISSUE TITLE
Cache requests to datagrepper

### ISSUE THREAD

**Issue by nphilipp** [fedora-infra/fmn#653]
Blocked by #714
~~Blocked by #621~~

# Story

As a user of the FMN API,
I want that data retrieved from datagrepper is appopriately cached,
so that interacting with the frontend is snappy.

# Acceptance Criteria

- [ ] Data retrieved from datagrepper is cached for a configurable lifetime

**Comment by ryanlerch** [comment]
@nphilipp is this one done now? can we close?

**Comment by nphilipp** [comment]
Hmm! We postponed this one at first because (unlike with Pagure and FASJSON) caching datagrepper queries probably won’t help much, it’s still a mountain of messages to wade through to guess how many notifications a new rule would produce per week. IIRC, we decided to rather track how many notifications a particular rule actually creates over time because we can count this “in passing” when the consumer has to process messages anyway.

That said (and /cc @abompard), I don’t think we need this anymore and should remove all the code accessing datagrepper (`get_last_messages()` etc.) because in its current state it takes ages to run making it practically useless. Any objections?
